org-code-javabuilder/lib/src/main/java/org/code/neighborhood/Grid.java

```diff
@@ -0,0 +1,49 @@
+package org.code.neighborhood;
+
+import java.util.ArrayList;
+
+public class Grid {
+  private GridSquare[][] grid;
+  private int width;
+  private int height;
+
+  protected Grid(GridSquare[][] squares) {
+    this.grid = squares;
+    this.height = squares.length;
+    this.width = squares[0].length;
+  }
+
+  public void printGrid() {
+    for (int y = height - 1; y >= 0; y--) {
+      ArrayList<String> squares = new ArrayList<String>();
+      for (int x = 0; x < width; x++) {
+        squares.add(grid[x][y].getPrintableDescription());
+      }
+      System.out.println(String.join(",", squares));
+    }
+  }
+
+  // Determines whether the given coordinate can be moved into
```

Comment:
NIT: Update the comment here to describe what would cause this to return true (or false). Reasoning - with the long chain of && conditions necessary here, it's easier to reason about what's happening if there's a "pretty" description.